modules/bidViewability.js

```diff
@@ -78,24 +78,21 @@ export let impressionViewableHandler = (globalModuleConfig, slot, event) => {
   }
 };
 
-export let init = () => {
-  events.on(EVENTS.AUCTION_INIT, () => {
-    // read the config for the module
-    const globalModuleConfig = config.getConfig(MODULE_NAME) || {};
-    // do nothing if module-config.enabled is not set to true
-    // this way we are adding a way for bidders to know (using pbjs.getConfig('bidViewability').enabled === true) whether this module is added in build and is enabled
-    if (globalModuleConfig[CONFIG_ENABLED] !== true) {
-      return;
-    }
-    // add the GPT event listener
-    window.googletag = window.googletag || {};
-    window.googletag.cmd = window.googletag.cmd || [];
-    window.googletag.cmd.push(() => {
-      window.googletag.pubads().addEventListener(GPT_IMPRESSION_VIEWABLE_EVENT, function(event) {
-        impressionViewableHandler(globalModuleConfig, event.slot, event);
-      });
+const handleSetConfig = (config) => {
+  const globalModuleConfig = config || {};
+  // do nothing if module-config.enabled is not set to true
+  // this way we are adding a way for bidders to know (using pbjs.getConfig('bidViewability').enabled === true) whether this module is added in build and is enabled
+  if (globalModuleConfig[CONFIG_ENABLED] !== true) {
+    return;
```

Comment:
The OP does not call this a bugfix, but I think the previous version would attach duplicate GPT event listeners on every auction. This fix, to be complete, should detach it here (when the module is turned off) IMO.

https://developers.google.com/publisher-tag/reference#googletag.Service.removeEventListener

Reply:
Made the changes as per the GPT document. where we are removing the listener before firing the handler.
Also consumed the _**event**_ object to extract the **_slot_** to handle Co-pilot's suggestion.